languages/elixir/concepts/lists/about.md

```diff
@@ -26,6 +35,15 @@ list = [2, 1]
 # => true
 ```
 
+Appending elements to a list during iteration is considered an anti-pattern. Appending an element requires walking through the entire list and adding the element at the end, therefore, appending a new element in each iteration would require walking through the entire list in each iteration.
+
+We can achieve the same result by prepending an element to the reversed list, and then reversing the result. Prepending is a fast operation and requires constant time.
+
+```elixir
+# [1, 2, 3] ++ [4, 5, 6] is equivalent to the much faster operation:
+Enum.reverse([6 | [5 | [4 | Enum.reverse([1, 2, 3])]]])
```

Comment:
For this particular example I don't think this holds true, I think what you mean is really:

`[1, 2, 3] ++ [4] ++ [5] ++ [6]` is equivalent.

```suggestion
# Appending to the end of a list (potentially slow)
[1, 2, 3] ++ [4] ++ [5] ++ [6]

# Prepend to the start of a list then reverse (faster, due to the nature of linked lists)
[6 | [5 | [4 | [3, 2, 1]]]] |> reverse()
```

Reply:
Yes, this is what I meant! Thanks for correcting.

Reply:
Just thinkjng more about this, rather than listing a fictional function called `reverse()` we could just say :
 `[6 | [5 | [4 | [3, 2, 1]]]] # then reverse`